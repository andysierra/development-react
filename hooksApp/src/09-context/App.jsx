import { Routes, Route } from 'react-router-dom';
import { Navbar } from '../components/Navbar';
import { HomePage } from './pages/HomePage';
import { Login } from './pages/Login';
import { About } from './pages/About';
import { NotFound404 } from './pages/NotFound404';
import { useState } from 'react';

export const App = () => {
  
  const [links, setLinks] = useState([
    {
      label:      'Home Page',
      to:         '/'
    },
    {
      label:      'Login',
      to:         'login'
    },
    {
      label:      'About',
      to:         'about'
    }
  ]);
  
  return (
    <>
      <h1 className="my-0">App</h1>
      <Navbar links={links} className="my-0 justify-content-center"/>
      <hr className="mt-0"/>
      <Routes>
        <Route path="/"       element={ <HomePage/>     } exact/>
        <Route path="login"   element={ <Login/>        } exact/>
        <Route path="about"   element={ <About/>        } exact/>

        <Route path="/*"      element={ <NotFound404/>  }/>
      </Routes>
    </>
  )
}