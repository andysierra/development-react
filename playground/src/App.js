import React from 'react';
import PropTypes from 'prop-types';

const App = ({
  name,
  apellido
}) => {

  return (
    <>
      <h1>Hola mundo React!</h1>
      <hr></hr>
      <p>Tu nombre es {name} y tu apellido es {apellido}</p>
    </>
  );
}

App.propTypes = {
  name: PropTypes.string.isRequired
};

App.defaultProps = {
  apellido: '[sin apellidos]'
}

export default App;