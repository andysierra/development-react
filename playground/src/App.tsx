import { UseStateTriggersARender } from "./components/UseStateTriggersARender/UseStateTriggersARender";
import { Padre } from './components/07-tarea-memo/Padre'; 

const App = () => {
  return (
    <div className="container-fluid w-100 h-100">

      <div className="row row-cols-3 p-2">
        <h1 className="display-6">Playground</h1>
      </div>

      <div className="row row-cols-3 g-2 py-2">
        <div className="col-12 col-sm-6 col-md-6 col-lg-4 col-xl-4">
          <div className="bg-danger p-1 h-100">
            <b>UseStateTriggersARender: No, only running outside</b>
            <p className="p-0 m-0">
              memorized component with React.memo + 
              props frozen with useCallback prevent inner component outside running</p>
            <small className="p-0 m-0">see console</small>
            <UseStateTriggersARender/>
          </div>
        </div>
        <div className="col-12 col-sm-6 col-md-6 col-lg-4 col-xl-4">
          <div className="bg-secondary p-1 h-100">
            <Padre/>
          </div>
        </div>
        <div className="col-12 col-sm-6 col-md-6 col-lg-4 col-xl-4">
          <div className="bg-warning p-1 h-100">
            <p>pez</p>
          </div>
        </div>
        <div className="col-12 col-sm-6 col-md-6 col-lg-4 col-xl-4">
          <div className="bg-danger p-1 h-100">
            <p>pez</p>
          </div>
        </div>
        <div className="col-12 col-sm-6 col-md-6 col-lg-4 col-xl-4">
          <div className="bg-secondary p-1 h-100">
            <p>pez</p>
          </div>
        </div>
        <div className="col-12 col-sm-6 col-md-6 col-lg-4 col-xl-4">
          <div className="bg-warning p-1 h-100">
            <p>pez</p>
          </div>
        </div>
        <div className="col-12 col-sm-6 col-md-6 col-lg-4 col-xl-4">
          <div className="bg-danger p-1 h-100">
            <p>pez</p>
          </div>
        </div>
        <div className="col-12 col-sm-6 col-md-6 col-lg-4 col-xl-4">
          <div className="bg-secondary p-1 h-100">
            <p>pez</p>
          </div>
        </div>
        <div className="col-12 col-sm-6 col-md-6 col-lg-4 col-xl-4">
          <div className="bg-warning p-1 h-100">
            <p>pez</p>
          </div>
        </div>
      </div>
    </div>
  )
}

export default App;