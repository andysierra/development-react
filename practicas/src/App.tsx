import { Navbar } from "./components/Navbar/Navbar";
import { NavbarLink } from "./components/Navbar/NavItems/NavbarLink";

const App = () => {

  return (
    <>
      <Navbar /*orientation*/>
        <NavbarLink path="/"      label="Practicas"       kind="brand"/>
        <NavbarLink path="/toDo"  label="ToDo"            kind="navLink"/>
        <NavbarLink path="/"      label="Future projects" kind="disabled"/>
      </Navbar>
    </>
  );
}

export default App;