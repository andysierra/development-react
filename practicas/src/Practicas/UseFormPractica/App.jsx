import React from 'react'
import { useForm } from './hooks/useForm'

export const App = () => {

  const {name, password, age, onChange, onClean} = useForm({
    name: "",
    password: "",
    age: 0
  });

  return (
    <div className="m-4 w-25">
      <h1>Pr√°ctica de forms</h1>
      <hr />

      <form className="form" method="POST">
        <div className="form-group">
          <label htmlFor="name">Nombre:</label>
          <input 
            type="text" 
            name="name" 
            className="form-control"
            value={name}
            onChange={onChange}
            placeholder="Nombre" />
        </div>
        <div className="form-group">
          <label htmlFor="password">Clave:</label>
          <input 
            type="password" 
            name="password" 
            className="form-control"
            value={password}
            onChange={onChange}
            placeholder="Clave" />
        </div>
        <div className="form-group">
          <label htmlFor="age">Edad:</label>
          <input 
            type="number" 
            name="age" 
            className="form-control"
            value={age}
            onChange={onChange}
            placeholder="Edad" />
        </div>
        <hr />
        <button 
          type="submit" 
          className="btn btn-sm btn-success mx-2"
          >Enviar</button>
        <button 
          type="button" 
          className="btn btn-sm btn-danger mx-2"
          onClick={onClean}>Limpiar</button>
      </form>

    </div>
  )
}
