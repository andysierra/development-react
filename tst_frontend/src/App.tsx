import { Form1 } from "./components/forms/Form1";

const App = () => {
  return (
    <Form1/>
  )
}

export default App;