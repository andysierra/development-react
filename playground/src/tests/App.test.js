/* import { render } from "@testing-library/react"; */
import App from "../App";
import {shallow} from 'enzyme';

describe('Pruebas de <App/>', () => {
  /* test('Debe mostrar mensaje "Tu nombre es ... y tu apellido es ..."', done => {
    
    const name = "Goku";
    const wrapper = render(<App name={name}/>);
    const txt = wrapper.findByText();

    txt.then(what=>{
      console.log(what);
      done();
    });
  }); */

  test('debe de mostrar <App/> correctamente', () => {
    const name      = 'Ionut';
    const apellido  = 'Rayannez';
    const saludo = <p>Tu nombre es {name} y tu apellido es {apellido}</p>;

    const wrapper = shallow(<App name={name} apellido={apellido}/>);
    
    //expect(wrapper.contains(saludo)).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  test('El name no debe ser vacÃ­o', () => {
    const name      = 'perro';
    
    const wrapper = shallow(<App name={name}/>);

    console.log(wrapper.props().name);
  });
})