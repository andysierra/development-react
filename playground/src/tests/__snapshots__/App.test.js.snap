// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <App/> debe de mostrar <App/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola mundo React!
  </h1>
  <hr />
  <p>
    Tu nombre es 
    Ionut
     y tu apellido es 
    Rayannez
  </p>
</Fragment>
`;
